crates/wasmtime/src/component/func/host.rs

```diff
@@ -330,3 +353,194 @@ macro_rules! impl_into_component_func {
 }
 
 for_each_function_signature!(impl_into_component_func);
+
+unsafe fn call_host_dynamic<T, F>(
```

Comment:
Similar to `TypedFunc::call` vs `Func::call`, is it possible to share this code with the existing entrypoint above? Given how different the parameters are represented it may be too onerous though.

Reply:
That was my first thought, but I couldn't see an elegant way to do it.  With `Func::call` I was able to reuse `call_raw` by using `[ValRaw: MAX_FLAT_PARAMS]` as the `LowerParams` generic parameter.  I can't use that trick with `call_host`, though, since the space for the parameters has already been allocated by the caller.

There probably is a way to refactor `call_host` to work in both contexts, but I expect it would be pretty ugly.

Reply:
Ok no worries, but in that case can you duplicate some of the more interesting test cases for imports for `func_new`-defined imports as well?